email-service/src/main/java/com/vgur/spring/email/services/EmailSending.java

```diff
@@ -2,52 +2,67 @@
 
 import com.vgur.spring.api.core.OrderDto;
 import com.vgur.spring.email.config.EmailProperties;
-import jakarta.annotation.PostConstruct;
 import lombok.RequiredArgsConstructor;
 
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.amqp.rabbit.annotation.RabbitListener;
-import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.mail.SimpleMailMessage;
 import org.springframework.mail.javamail.JavaMailSender;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 @Service
 @RequiredArgsConstructor
 @Slf4j
-@EnableConfigurationProperties(
-        EmailProperties.class
-)
 public class EmailSending {
     private final JavaMailSender mailSender;
     private final EmailProperties emailProperties;
-    private ArrayList<OrderDto> ordersToSend;
-
-    @PostConstruct
-    public void init(){
-        ordersToSend = new ArrayList<>();
-    }
+    private ArrayList<OrderDto> ordersToSend = new ArrayList<>();
 
     @RabbitListener(queues = "EmailSending")
-    public void listenEmailSendingQueue(OrderDto orderDto){
+    public void listenEmailSendingQueue(OrderDto orderDto) {
         ordersToSend.add(orderDto);
     }
 
-    @Scheduled(fixedDelayString ="${spring.mail.sending_delay}" )
-    private synchronized void sendEmails(){
+    @Scheduled(fixedDelayString = "${spring.mail.sending_delay}")
+    private void sendEmails() {
         var copyOrders = (OrderDto[]) ordersToSend.toArray();
         ordersToSend.clear();
-        for (OrderDto orderDto: copyOrders) {
-            sendSimpleMessage(orderDto.getEmail(),"new order","You make new order");
+
+        int threadQuantity = 4;
```

Comment:
why did u choose 4 ? 
FYI u can use next option to programmatically determine amount of procrssors :         int availableProcessors = Runtime.getRuntime().availableProcessors();